### ISSUE TITLE
staging job is failing with dpkg lock contention

### ISSUE THREAD

**Issue by legoktm** [freedomofpress/securedrop#7258]
* https://github.com/freedomofpress/securedrop/actions/runs/11377625042/attempts/1

`failed: [mon-staging] (item=securedrop-keyring_0.2.2+2.10.0+focal_all.deb) => {"ansible_loop_var": "item", "changed": true, "cmd": ["dpkg", "-i", "/root/securedrop-keyring_0.2.2+2.10.0+focal_all.deb"], "delta": "0:00:00.007442", "end": "2024-10-17 03:28:49.153608", "item": "securedrop-keyring_0.2.2+2.10.0+focal_all.deb", "msg": "non-zero return code", "rc": 2, "start": "2024-10-17 03:28:49.146166", "stderr": "dpkg: error: dpkg frontend lock is locked by another process", "stderr_lines": ["dpkg: error: dpkg frontend lock is locked by another process"], "stdout": "", "stdout_lines": []}`

* https://github.com/freedomofpress/securedrop/actions/runs/11377625042/attempts/2

`failed: [mon-staging] (item=securedrop-ossec-server_3.6.0+2.10.0+focal_all.deb) => {"ansible_loop_var": "item", "changed": true, "cmd": ["dpkg", "-i", "/root/securedrop-ossec-server_3.6.0+2.10.0+focal_all.deb"], "delta": "0:00:00.008102", "end": "2024-10-17 15:21:57.993267", "item": "securedrop-ossec-server_3.6.0+2.10.0+focal_all.deb", "msg": "non-zero return code", "rc": 2, "start": "2024-10-17 15:21:57.985165", "stderr": "dpkg: error: dpkg frontend lock is locked by another process", "stderr_lines": ["dpkg: error: dpkg frontend lock is locked by another process"], "stdout": "", "stdout_lines": []}
failed: [mon-staging] (item=ossec-server_3.6.0+focal_amd64.deb) => {"ansible_loop_var": "item", "changed": true, "cmd": ["dpkg", "-i", "/root/ossec-server_3.6.0+focal_amd64.deb"], "delta": "0:00:00.007470", "end": "2024-10-17 15:21:58.304197", "item": "ossec-server_3.6.0+focal_amd64.deb", "msg": "non-zero return code", "rc": 2, "start": "2024-10-17 15:21:58.296727", "stderr": "dpkg: error: dpkg frontend lock is locked by another process", "stderr_lines": ["dpkg: error: dpkg frontend lock is locked by another process"], "stdout": "", "stdout_lines": []}`

Not sure if this is related to the molecule upgrade or a race condition we've just exposed.

**Comment by cfm** [comment]
This is now happening in [`hold_debs.yml`](https://github.com/freedomofpress/securedrop/blob/96b4717baa49312b104961b59aad540144027cd3/install_files/ansible-base/roles/install-local-packages/tasks/hold_debs.yml#L13-L17) ([e.g.](https://github.com/freedomofpress/securedrop/actions/runs/14987452095/job/42103936911#step:3:38375)).

**Comment by legoktm** [comment]
tbh I wonder if this new failure is the noble migration kicking off as soon as securedrop-config is installed, especially since we just merged the `noble-upgrade.json` changes into develop.

**Comment by eloquence** [comment]
@cfm @legoktm Have we seen this failure on Noble as well? Otherwise I suggest we close this in favor of #7562 (which tracks Focal-specific failures).

**Comment by legoktm** [comment]
I don't think so - or if it has flakily appeared, it's a different root cause than what we fixed here. (Correct me if I'm wrong Cory)
